guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,954 @@
+
+### Example
+
+```rust
+// File: src/models.rs
+
+use schema::users; // Brings the users table into scope
+
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+#[derive(Queryable)]
+pub struct EmailUser {
+    pub id: i32,
+    pub email: String,
+}
+```
+
+```rust
+// File: src/main.rs
+#[macro_use] extern crate diesel;
+
+use diesel::prelude::*;
+
+fn main() {
+    // The following will return all Users as a QueryResult<Vec<User>>
+    let users_result: QueryResult<Vec<User>> = users.load(&db_connection);
+
+    // Here we are getting the value (or error) out of the QueryResult
+    // A successful value will be of type Vec<User>
+    let users = users_result.expect("Error loading users");
+
+    // Here, a successful value will be type Vec<EmailUser> 
+    let email_users = users.select((users::id, users::email))
+        .load::<EmailUser>(&db_connection)
+        .expect("Error loading the email only query");
+}
+```
+
+If we were to comment out all the three `String` fields on our `User` struct, we would see the following error.
+
+```rust
+error[E0277]: the trait bound `(i32,): diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not satisfied
+  --> src/bin/main.rs:34:10
+   |
+34 |         .load::<User>(&db_conn)
+   |          ^^^^ the trait `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not implemented for `(i32,)`
+   |
+   = help: the following implementations were found:
+             <(A,) as diesel::Queryable<(SA,), DB>>
+   = note: required because of the requirements on the impl of `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` for `diesel_demo::models::User`
+   = note: required because of the requirements on the impl of `diesel::LoadQuery<_, diesel_demo::models::User>` for `diesel_demo::schema::users::table`
+```
+
+Notice the compiler is indicating a trait is not implmented for calling the `.load()` method.
+When reading, take note of the values in the tuple[s].
+
+> `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>`
+
+
+`Queryable` is trying to convert those 4 types into the types on our `User` model struct.
+Our model struct has the three String fields commented out, so it doesn't know what those `Text` columns are supposed to be converted to.
+Remember, `Queryable` structs represent the exact columns, value, 
+and ordering of your query's returned result,
+which we are violating here.
+`Queryable` is expecting a tuple that looks like `(i32, String, String, String)`,
+but we currently only have a tuple consisting of `(i32,)`.
+We need to add those `String` columns back for our code to compile again.
+See Rust's [data type docs] if you're unfamiliar with tuples or their syntax.
```

Comment:
I'd add this sentence in parenthesis (sadly we don't have footnotes here) the first time you use/refer to a tuple :)